src/helpers/radix_sort.rs

```diff
@@ -0,0 +1,247 @@
+use crate::modulus_convert::make_three;
+use crate::modulus_convert::multiply_secret_shares;
+use crate::modulus_convert::RandomShareGenerationHelper;
+use crate::modulus_convert::ReplicatedFp31SecretSharing;
+
+type ReplicatedShares = (
+    ReplicatedFp31SecretSharing,
+    ReplicatedFp31SecretSharing,
+    ReplicatedFp31SecretSharing,
+);
+
+type RandomShareGenerationHelpers = (
+    RandomShareGenerationHelper,
+    RandomShareGenerationHelper,
+    RandomShareGenerationHelper,
+);
+
+#[must_use]
+pub fn mpc_sort(shared_secrets: &[ReplicatedShares]) -> Vec<ReplicatedShares> {
+    let helpers = make_three();
+    let (helpers, mult_inputs) = helpers_gen_mult_inputs(helpers, shared_secrets);
+    let (_, mult_output) = helpers_do_multiplication(helpers, &mult_inputs);
+    helpers_gen_permutations(&mult_output)
+}
+
+// Wrapper functions over helper functions
+fn helpers_gen_mult_inputs(
+    helpers: RandomShareGenerationHelpers,
+    shared_secrets: &[ReplicatedShares],
+) -> (
+    RandomShareGenerationHelpers,
+    Vec<Vec<(ReplicatedFp31SecretSharing, ReplicatedFp31SecretSharing)>>,
+) {
+    let shares = get_helper_specific_shares(shared_secrets);
+    let (h1, h2, h3) = helpers;
+
+    let mut mult_inputs = vec![helper_gen_mult_input(&shares[0], &h1)];
+    mult_inputs.push(helper_gen_mult_input(&shares[1], &h2));
+    mult_inputs.push(helper_gen_mult_input(&shares[2], &h3));
+
+    ((h1, h2, h3), mult_inputs)
+}
+
+fn helpers_do_multiplication(
+    helpers: RandomShareGenerationHelpers,
+    mult_inputs: &[Vec<(ReplicatedFp31SecretSharing, ReplicatedFp31SecretSharing)>],
+) -> (RandomShareGenerationHelpers, Vec<ReplicatedShares>) {
+    let (h1, h2, h3) = helpers;
+    let mut mult_outputs = Vec::new();
+    (0..mult_inputs[0].len()).for_each(|i| {
+        let a1 = mult_inputs[0][i].0;
+        let a2 = mult_inputs[1][i].0;
+        let a3 = mult_inputs[2][i].0;
+        let b1 = mult_inputs[0][i].1;
+        let b2 = mult_inputs[1][i].1;
+        let b3 = mult_inputs[2][i].1;
+        mult_outputs.push(multiply_secret_shares(
+            &h1, &h2, &h3, a1, a2, a3, b1, b2, b3,
+        ));
```

Comment:
I feel like it would have been more clear to just do step 1 of the multiplication within the helpers_gen_mult_inputs function. Having that function just return a data structure with all the inputs, just to pass them into this function which has to re-assemble all the inputs out of that data structure doesn't add anything but complexity.